plugins/plugin-java-debugger/che-plugin-java-debugger-server/src/main/java/org/eclipse/che/plugin/jdb/server/JavaDebugger.java

```diff
@@ -698,21 +687,47 @@ private void clearSteps() throws DebuggerException {
 
   @Override
   public String evaluate(String expression) throws DebuggerException {
-    com.sun.jdi.Value result = evaluate(ExpressionParser.newInstance(expression));
+    lock.lock();
+    try {
+      return evaluate(expression, getCurrentThread().uniqueID(), 0);
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  @Override
+  public String evaluate(String expression, long threadId, int frameIndex)
```

Comment:
The method above use lock to prevent simultaneous execution of this one. What if someone call this public method directly?